src/Polynomials/MonomialBases.jl

```diff
@@ -116,15 +116,16 @@ end
 return_type(::MonomialBasis{D,T}) where {D,T} = T
 
 # Field implementation
-
 function return_cache(f::MonomialBasis{D,T},x::AbstractVector{<:Point}) where {D,T}
   @assert D == length(eltype(x)) "Incorrect number of point components"
+  xi = testitem(x)
+  Tp = typeof(zero(T)*zero(eltype(eltype(x))))
```

Comment:
Perhaps this is a bit more general. I think that you need to find the type of the linear combination of monomials

```
zT = zero(T)
zxi = zero(eltype(eltype(x))
Tp = typeof( zT*zxi*zxi + zT*zxi*zxi  )
```

Reply:
thanks francesc, I have made the update.